modules/login/src/main/resources/META-INF/resources/login/js/registration.js

```diff
@@ -0,0 +1,401 @@
+var registration = function($) {
+    var orgLookupData = {};
+    var bInitialized = false;
+
+    var $formStep1, $formStep2,$formStep3,$formStep4; //Form selector cache.
+    var $dlgStep1, $dlgStep2,$dlgStep3,$dlgStep4,$dlgConfirmation,$dlgError; //Dialog selector cache.
+    var $selectOrgType, $selectOrg, $selectImtUSAR, $selectImtFed, $selectImtCDF, $selectImtLocal; //Select selector cache.
+
+    var endpointRoot = "";
+    var orgtypesService = "organizationTypes";
+    var orgsService = "organizations";
+    var registrationService = "register";
+    var termsUrl = "login/html/terms.html";
+    var termsElement = "#tc_content";
+
+    function setEndPointRoot(root) {
+        endpointRoot = root;
+    }
+
+    function setTermsUrl(url) {
+        termsUrl = url;
+    }
+
+    function init() {
+
+        if (!bInitialized) {
+            //Selector cache
+            $formStep1 = $('#formStep1');
+            $formStep2 = $('#formStep2');
+            $formStep3 = $('#formStep3');
+            $formStep4 = $('#formStep4');
+
+            $dlgStep1 = $('#step1');
+            $dlgStep2 = $('#step2');
+            $dlgStep3 = $('#step3');
+            $dlgStep4 = $('#step4');
+            $dlgConfirmation = $('#regConfirmation');
+            $dlgError = $('#errordialog');
+
+            $selectOrgType = $('#selectOrgAffiliates');
+            $selectOrg = $('#selectOrganizations');
+
+            $selectImtCDF = $('#selectCDFIMT');
+            $selectImtFed = $('#selectFedIMT');
+            $selectImtLocal = $('#selectLocalIMT');
+            $selectImtUSAR = $('#selectUSAR');
+
+            $($formStep1).validator().on('submit', doStep2);
+
+            $($formStep2).validator().on('submit', doStep3);
+            //Listen for field validation events on Form2 in order to detect a valid email address
+            // was entered and send to the verify service.
+            //$($formStep2).validator().on('validated.bs.validator', form2OnValidateField);
+
+            $($formStep3).validator().on('submit', doStep4);
+            $($formStep4).validator().on('submit', doRegistration);
+            $($formStep1).on('reset', onResetForm1);
+            $($formStep2).on('reset', onResetForm2);
+            $($formStep3).on('reset', onResetForm3);
+            $($formStep4).on('reset', onResetForm4);
+
```

Comment:
More preference here, but the use of whitespace is inconsistent and confusing. Are you trying to communicate grouping? If so, consider using a function so that you can effectively name the group instructions.